tests/ocp/sriov/README.md

```diff
@@ -0,0 +1,557 @@
+    ContinueOnFailure, 
+    func() {
+        var (
+            // Shared test variables
+        )
+
+        BeforeAll(func() {
+            // Setup required for all tests in this Describe block
+            By("Setting up test environment")
+            // ... setup code ...
+        })
+
+        AfterAll(func() {
+            // Cleanup after all tests
+            By("Cleaning up test resources")
+            // ... cleanup code ...
+        })
+
+        BeforeEach(func() {
+            // Setup for each test
+        })
+
+        AfterEach(func() {
+            // Cleanup after each test
+        })
+
+        It("should perform specific test action", reportxml.ID("12345"), func() {
+            By("Step description")
+            // Test implementation
+        })
+    })
+```
+
+#### Test Organization Principles
+
+1. **Ordered Tests**: Use `Ordered` when tests must run in sequence
+2. **Labels**: Always use labels for test filtering (`Label(tsparams.LabelFeatureName)`)
+3. **ContinueOnFailure**: Use when appropriate to gather more diagnostic information
+4. **Descriptive Names**: Test descriptions should clearly state what is being tested
+
+### Test Labels
+
+#### Label Naming Convention
+
+- Use lowercase with hyphens or underscores
+- Be specific and descriptive
+- Group related tests under the same label
+- Define labels in `internal/tsparams/consts.go`
+
+Example:
+```go
+const (
+    LabelSuite = "sriov"
+    LabelFeatureName = "feature-name"
+    LabelHWEnabled = "sriov-hw-enabled"
+)
+```
+
+### Test IDs
+
+- Use `reportxml.ID("12345")` for each test case
+- Test IDs should be unique and traceable to test case management systems
+- Include test ID in test description or comments when helpful
+
+## Internal Packages
+
+### Test Parameters Package (`tsparams`)
+
+Purpose: Centralize constants, variables, and configuration for the test suite.
+
+#### Structure
+
+```go
+package tsparams
+
+const (
+    // Labels
+    LabelSuite = "sriov"
+    LabelFeatureName = "feature-name"
+    
+    // Namespaces
+    TestNamespaceName = "sriov-tests"
+    
+    // Timeouts
+    WaitTimeout = 3 * time.Minute
+    DefaultTimeout = 300 * time.Second
+)
+
+var (
+    // Labels list for test selection
+    Labels = []string{LabelSuite, LabelFeatureName}
+    
+    // Reporter configuration
+    ReporterCRDsToDump = []k8sreporter.CRData{...}
+    ReporterNamespacesToDump = map[string]string{...}
+)
+```
+
+#### Best Practices
+
+1. **Constants**: Use for values that don't change (labels, namespaces, default values)
+2. **Variables**: Use for values that may be configured or computed
+3. **Timeouts**: Define reasonable defaults; make them configurable when needed
+4. **Labels**: Aggregate in a `Labels` variable for suite-level filtering
+
+### Environment Validation Package (`sriovenv` or similar)
+
+Purpose: Validate cluster state and test prerequisites.
+
+#### Responsibilities
+
+- Verify operator deployment status
```

Comment:
No objection on this line. Just a general comment that we should not assume sriov operator is deployed via OLM (e.g., catsrc, subs, csv..). Need to also support pre-merge CI type of use case or operator deployed via make deploy, etc.